lib/ios-log.js

```diff
@@ -0,0 +1,217 @@
+import path from 'path';
+import _ from 'lodash';
+import which from 'which';
+import glob from 'glob';
+import { fs, getLogger } from 'appium-logger';
+import mkdirp from 'mkdirp';
+import touch from 'touch';
+import xcode from 'xcode'; 
+import { SubProcess } from 'teen_process';
+
+// Date-Utils: Polyfills for the Date object
+
+let START_TIMEOUT = 10000;
+let logger = getLogger('IOS-log');
+
+class IOSLog { 
+
+  constructor(opts) {
+    this.udid = opts.udid;
+    this.simUdid = opts.simUdid;
+    this.showLogs = opts.showLogs;
+    this.proc = null;
+    this.onIosLogStart = null;
+    this.iosLogStarted = false;
+    this.iosLogStartTime = null;
+    this.loggingModeOn = true;
+    this.logs = [];
+    this.logRow = '';
+    this.logsSinceLastRequest = [];
+    this.startTimer = null;
+  }
+
+  async startCapture () {
+    if (this.udid) { 
+      this.loggingModeOn = false;
+      let spawnEnv = _.clone(process.env);
+      logger.debug('Attempting iOS device log capture via libimobiledevice idevicesyslog');
+      let err = await which('idevicesyslog');
+      if (!err) { 
+        this.proc = new SubProcess('idevicesyslog', {env: spawnEnv});
+      } else {
+        logger.warn('Could not capture device log using libimobiledevice idevicesyslog.' +
+            ' Libimobiledevice probably is not installed');
+        logger.debug('Attempting iOS device log capture via deviceconsole');
+        let limDir = path.resolve(__dirname, '../../../build/deviceconsole');  //??? 
+        spawnEnv.PATH = `${process.env.PATH}:${limDir}`;
+        spawnEnv.DYLD_LIBRARY_PATH = `${limDir}:${process.env.DYLD_LIBRARY_PATH}`;
+        this.proc =  new SubProcess('deviceconsole', ['-u', this.udid], {env: spawnEnv});
+      }
+      this.finishStartingLogCapture();
+      return;
+    } 
+
+    let xcodeVersion = await xcode.getVersion();
+    let ver = parseInt(xcodeVersion.split('.')[0], 10);
+
+    if ( ver < 5 ) {
+      logger.debug('Starting iOS6.* simulator log capture');
+      this.proc = new SubProcess('tail', ['-f', '-n', '1', '/var/log/system.log']);
+      this.finishStartingLogCapture();
+      return;
+    }
+
+    let logsPath;
+
+    if (ver >= 6) { 
+      logger.debug('Starting iOS 8.* simulator log capture');
```

Comment:
We would soon release 9.* support to so can we make this iOS 8+ or some thing which indicates iOS 8 and iOS 9. What do you think @jlipps ?